### PR TITLE
Ensure we validate against correct `SPOClassUID` in `pixl_dcmd` tests

### LINKED ISSUE #580: Ensure we're validating against correct modalities in `pixl_dcmd` tests
### Definition of Done / Acceptance Criteria

Currently, when validating the anonymised DICOMs in the `pixl_dcmd` tests, we set the modality of the generated dataset but not the `SOPClassUID`.

As `dicom-validator` [validates the DICOM based on the `SOPClassUID`](https://github.com/pydicom/dicom-validator/blob/27234b418d17491e913c0b6e79136da35fb2181f/dicom_validator/validator/iod_validator.py#L130-L139) rather than modality, we are always validating against the [default for the generated dataset](https://github.com/SAFEHR-data/PIXL/blob/6435ebbb0e7990bfde42052b6fb292b2fa4100c4/pytest-pixl/src/pytest_pixl/dicom.py#L153) (which corresponds to an MR image).

- [x] We anonymise and validate against all modalities listed in all project configs

### Testing

Currently, if we set the `SOPClassUID` based on modality, the following validation errors are raised:
```python
FAILED tests/test_dicom_validator.py::test_validation_after_anonymisation_works - AssertionError: assert not {'DX Image': {'Tag (2050,0020) (Presentation LUT Shape) is missing'}}
FAILED tests/test_main.py::test_anonymise_and_validate_dicom[test-extract-uclh-omop-cdm] - assert 'WARNING' not in ['WARNING']
FAILED tests/test_main.py::test_anonymise_and_validate_dicom[test-extract-uclh-omop-cdm-xnat] - assert 'WARNING' not in ['WARNING']
FAILED tests/test_main.py::test_anonymise_and_validate_dicom[uclh-nasogastric-tube-project-ngt-only-full-dataset] - assert 'WARNING' not in ['WARNING']
FAILED tests/test_main.py::test_anonymise_and_validate_dicom[test-extract-uclh-omop-cdm-dicomweb] - assert 'WARNING' not in ['WARNING']
```
The tests should be updated to pass.

### Documentation

_No response_

### Dependencies

_No response_

### Details and Comments

_No response_

### PR DESCRIPTION
**Pull request by p-j-smith**
## Description
Fixes #580 

- add `MODALITY_TO_CLASS_UID` dictionary to `pytest_pixl.dicom` module to map DICOM modality to SOPClassUID
- update `test_anonymise_and_validate_dicom` to ensure we validate against all modalities listed in all project configs
- add missing tags to base config files (some were missing but not picked up as we weren't validating against the correct modalities / SOPClassUID)
- move duplicated tags into `base.yaml` (otherwise `test_merge_base_only_tags` complains that tags are duplicated across config files) 
- remove unused fixtures in `pixl_dcmd` tests (some tests were including fixtures that were not actually used / needed for the tests)
- rename `vanilla_single_dicom_image_DX` to `vanilla_dicom_image_DX` as the previous `vanilla_dicom_image_DX` fixture has been removed as not needed

## Type of change
Please delete options accordingly to the description.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Suggested Checklist

- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have passed on my local host device. (see further details at the [CONTRIBUTING](https://github.com/SAFEHR-data/PIXL/blob/main/CONTRIBUTING.md#local-testing) document)
- [ ] Make sure your branch is up-to-date with main branch. See [CONTRIBUTING](https://github.com/SAFEHR-data/PIXL/blob/main/CONTRIBUTING.md) for a general example to syncronise your branch with the `main` branch.
- [ ] I have requested review to this PR.
- [ ] I have addressed and marked as resolved all the review comments in my PR.
- [ ] Finally, I have selected `squash and merge`

### CODE REVIEW

File: pixl_dcmd/tests/conftest.py [hunk 1]
```diff
@@ -126,67 +126,6 @@ def row_for_testing_image_with_pseudo_patient_id(
...
-def ids_for_parameterised_test(val: pathlib.Path) -> str:
-    """Generate test ID for parameterised tests"""
-    return str(val.stem)
-
-
-@pytest.fixture()
-@pytest.mark.parametrize(
-    ("yaml_file"), PROJECT_CONFIGS_DIR.glob("*.yaml"), ids=ids_for_parameterised_test
-)
-def row_for_dicom_testing(db_session, yaml_file) -> Session:
```
Comment (p-j-smith) [inline pixl_dcmd/tests/conftest.py:138]: this is not needed (any tests that need the DB already use the `rows_in_session` fixture instead)

File: pixl_dcmd/tests/conftest.py [hunk 2]
```diff
@@ -126,67 +126,6 @@ def row_for_testing_image_with_pseudo_patient_id(
...
-    )
-    with db_session:
-        db_session.add_all([extract, image_not_exported])
-        db_session.commit()
-
-    return db_session
-
-
-@pytest.fixture()
-def row_for_single_dicom_testing(db_session) -> Session:
```
Comment (p-j-smith) [inline pixl_dcmd/tests/conftest.py:166]: same for this fixture

File: pixl_dcmd/tests/conftest.py [hunk 3]
```diff
@@ -279,41 +218,15 @@ def mock_get(key, default) -> Optional[str]:
...
     Private tags were added to match the tag operations defined in the project config, so we can
     test whether the anonymisation process works as expected when defining overrides.
-    The row_for_single_dicom_testing dependency is to make sure the database is populated with the
-    project slug, which is used to anonymise the DICOM image.
     """
     return generate_dicom_dataset(Modality="DX")
 
 
-@pytest.fixture()
-def vanilla_dicom_image_MR(row_for_dicom_testing) -> Dataset:
```
Comment (p-j-smith) [inline pixl_dcmd/tests/conftest.py:306]: `generate_dicom_dataset` is now called directly in `test_anonymise_and_validate_dicom`, rather than creating fixtures for all different modalities we will support

File: pixl_dcmd/tests/test_main.py [hunk 4]
```diff
@@ -71,7 +71,7 @@ def _get_mri_diffusion_tags(
     so we can test whether the manufacturer overrides work during anonymisation
     """
     tag_ops = load_tag_operations(config)
-    mri_diffusion_overrides = tag_ops.manufacturer_overrides[0]
+    mri_diffusion_overrides = tag_ops.manufacturer_overrides[1]
```
Comment (p-j-smith) [inline pixl_dcmd/tests/test_main.py:74]: I've included `"mri.yaml"` in the manufacturer overrides for `test-extract-uclh-omop-cdm.yaml` (to be consistent with what's currently in `test-extract-uclh-omop-cdm-xnat.yaml` and `test-extract-uclh-omop-cdm-dicomweb.yaml`)

File: pixl_dcmd/tests/test_tag_schemes.py [hunk 5]
```diff
@@ -39,7 +40,7 @@ def test_merge_base_only_tags(base_only_tag_scheme):
     THEN the result should be the same as the base file
     """
     tags = merge_tag_schemes(base_only_tag_scheme)
-    expected = [*base_only_tag_scheme.base[0], *base_only_tag_scheme.base[1]]
+    expected = [tag for base in base_only_tag_scheme.base for tag in base]
```
Comment (p-j-smith) [inline pixl_dcmd/tests/test_tag_schemes.py:43]: there's three base tag schemes now, so including them all here

File: projects/configs/tag-operations/test-extract-uclh-omop-cdm.yaml [hunk 6]
```diff
@@ -14,11 +14,4 @@
...
 
-- name: "VR OB Creator"
-  group: 0x0013
-  element: 0x0010
-  op: "keep"
-- name: "VR OB sequence"
-  group: 0x0013
-  element: 0x1010
-  op: "replace"
+[]
```
Comment (p-j-smith) [inline projects/configs/tag-operations/test-extract-uclh-omop-cdm.yaml:17]: these have moved to `base.yaml` as they were duplicated in `xray.yaml`. I guess we could remove this file?

Comment (stefpiatek) [reply projects/configs/tag-operations/test-extract-uclh-omop-cdm.yaml:17]: Yeah happy with that

File: pixl_dcmd/tests/test_tag_schemes.py [hunk 7]
```diff
@@ -49,9 +50,9 @@ def test_merge_base_only_tags(base_only_tag_scheme):
             count_tags[key] = 1
 
     for key, values in count_tags.items():
-        assert (
-            values == 1
-        ), f"{key} is replicated please check config files to remove it"
+        pytest_check.equal(
+            values, 1, msg=f"{key} is replicated please check config files to remove it"
+        )
```
Comment (stefpiatek) [inline pixl_dcmd/tests/test_tag_schemes.py:55]: ooh interesting, why the change?

File: projects/configs/tag-operations/uclh-nasogastric-tube-project-ngt-only-full-dataset.yaml
Comment (stefpiatek) [inline projects/configs/tag-operations/uclh-nasogastric-tube-project-ngt-only-full-dataset.yaml:1]: could probably remove this file tbf

File: pytest-pixl/src/pytest_pixl/dicom.py [hunk 8]
```diff
@@ -62,6 +62,15 @@ def write_volume(filename_pattern: str) -> None:
...
 }
 
+# Mapping based on: https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_b.5.html
+MODALITY_TO_CLASS_UID = {
+    "CR": "1.2.840.10008.5.1.4.1.1.1",
+    "CT": "1.2.840.10008.5.1.4.1.1.2",
+    "DX": "1.2.840.10008.5.1.4.1.1.1.1",
+    "MR": "1.2.840.10008.5.1.4.1.1.4",
+    "PT": "1.2.840.10008.5.1.4.1.1.128",
+}
```
Comment (stefpiatek) [inline pytest-pixl/src/pytest_pixl/dicom.py:72]: 🎉

File: pytest-pixl/src/pytest_pixl/dicom.py [hunk 9]
```diff
@@ -111,6 +120,9 @@ def generate_dicom_dataset(tag_values: dict = TAGS_DICT, **kwargs: Any) -> Datas
             msg = f"Tag {key} is not a valid DICOM tag"
             raise ValueError(msg)
 
+    if "Modality" in kwargs and "SOPClassUID" not in kwargs:
+        ds.SOPClassUID = MODALITY_TO_CLASS_UID[kwargs["Modality"]]
```
Comment (stefpiatek) [inline pytest-pixl/src/pytest_pixl/dicom.py:124]: interesting that this isn't covered by a test. is coverage just playing up?

**Review by stefpiatek — APPROVED** [review]
Thanks for spotting and fixing Paul!

File: pixl_dcmd/tests/test_tag_schemes.py (hunk 7 above)
Comment (p-j-smith) [reply pixl_dcmd/tests/test_tag_schemes.py:55]: oh just so that if there are several duplicated tags, we will get a list of all of them from running the test once (previously the test would fail after the first failed assert, you'd remove the duplicated tag, then it might fail again on another tag)

File: pytest-pixl/src/pytest_pixl/dicom.py (hunk 9 above)
Comment (p-j-smith) [reply pytest-pixl/src/pytest_pixl/dicom.py:124]: hmm, yeah I think it must be covered as without it the tests fail

File: projects/configs/tag-operations/uclh-nasogastric-tube-project-ngt-only-full-dataset.yaml
Comment (p-j-smith) [reply projects/configs/tag-operations/uclh-nasogastric-tube-project-ngt-only-full-dataset.yaml:1]: ah okay, should the project config be removed too, or just the tag operation file?

Comment (stefpiatek) [reply projects/configs/tag-operations/uclh-nasogastric-tube-project-ngt-only-full-dataset.yaml:1]: I'd say the tag operation file and we use the xray one in its place

File: pixl_dcmd/tests/test_tag_schemes.py (hunk 7 above)
Comment (stefpiatek) [reply pixl_dcmd/tests/test_tag_schemes.py:55]: ah yes of course

File: projects/configs/tag-operations/uclh-nasogastric-tube-project-ngt-only-full-dataset.yaml
Comment (p-j-smith) [reply projects/configs/tag-operations/uclh-nasogastric-tube-project-ngt-only-full-dataset.yaml:1]: yeah that makes sense

Comment (p-j-smith) [reply projects/configs/tag-operations/uclh-nasogastric-tube-project-ngt-only-full-dataset.yaml:1]: I've moved all the tags from this file into either the xray tag config file or the base config file
